pyarrow-stubs/_flight.pyi

```diff
@@ -83,6 +83,9 @@ class BasicAuth(_Weakrefable):
     def username(self) -> bytes: ...
     @property
     def password(self) -> bytes: ...
+    def serialize(self) -> str: ...
+    @staticmethod
+    def deserialize(serialized: str | bytes) -> BasicAuth: ...
```

Comment:
[source link](https://github.com/apache/arrow/blob/77f70e4a122c50f28d9ff97debb8831d2558f4d3/python/pyarrow/_flight.pyx#L413-L421) - the types aren't annotated but can be inferred. `deserialize` calls `tobytes` on its input like the constructor, and `serialize` calls `SerializeToString` (and `GetResultValue` appears to just move the value